src/lineup/internal/GeneralVisWrapper.ts

```diff
@@ -123,25 +75,61 @@ export class GeneralVisWrapper extends EventHandler {
     }
 
     updateCustomVis() {
-        const data = this.getAllData();
-        const colDescriptions = this.provider.getColumns();
-        //need some way to convert these to _ids.
-        const cols: any[] = [];
+        const ranking = this.provider.getFirstRanking();
+
+        const data = this.provider.viewRawRows(ranking.getOrder());
+
+        const cols: VisColumn[] = [];
 
         const selectedMap: { [key: number]: boolean } = this.getSelectionMap();
 
-        for(const c of colDescriptions.filter((d) => d.type === 'number' || d.type === 'categorical')) {
-            cols.push({
-                info: {
-                    name: c.label.replace(/(<([^>]+)>)/gi, ''),
-                    description: c.summary ? c.summary.replace(/(<([^>]+)>)/gi, '') : '',
-                    id: c.label + (<any> c)._id
-                },
-                values: data.map((d, i) => {
-                    return {id: d._id, val: d[(<any> c).column] ? d[(<any> c).column] : c.type === 'number' ? null : '--'};
-                }),
-                type: c.type === 'number' ? EColumnTypes.NUMERICAL : EColumnTypes.CATEGORICAL
+        const getColumnInfo = (column: Column): ColumnInfo => {
+            return {
+                name: column.getMetaData().label.replace(/(<([^>]+)>)/gi, ''),
+                description: column.getMetaData().summary.replace(/(<([^>]+)>)/gi, ''),
+                // TODO: What kind of id to use?
+                id: column.fqid,
+            };
+        };
+
+
+        // wait for 2 seconds
+
+        const getColumnValue = async <T,>(column: ValueColumn<T>) => {
+
+            if (column.isLoaded()) {
+                return data.map((d, i) => ({id: (<IRow>d.v)._id, val: column.getValue(d)}));
+            }
+
+            return new Promise<{id: number, val: T}[]>((resolve, reject) => {
+                const timeout = setTimeout(() => {
+                    reject('Timeout');
+                }, 60000);
+
+                column.on(ValueColumn.EVENT_DATA_LOADED, () => {
+                    clearTimeout(timeout);
+                    resolve(data.map((d, i) => ({id: (<IRow>d.v)._id, val: column.getValue(d)})));
+                });
             });
+        };
+
+        for(const c of ranking.flatColumns) {
+            if(c instanceof NumberColumn) {
+                cols.push({
+                    info: getColumnInfo(c),
+                    values: () => getColumnValue(c),
+                    type: EColumnTypes.NUMERICAL
+                });
+            }
+            if(c instanceof CategoricalColumn) {
+                cols.push({
+                    info: getColumnInfo(c),
+                    values: () => getColumnValue(c).then((res) => res.map((v) => v.val ? v : {...v, val: '--'})),
+                    // TODO: This is required?
+                    colors: null,
```

Comment:
If no value is ever assigned to `colors` it could be removed. Or what does this TODO mean?